sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -78,6 +80,8 @@ def _send_request_helper(self, url, accepted_status_codes, function, data, retry
                 error = False
                 status = r.status_code
                 text = r.text
+                if self._endpoint.auth == constants.AUTH_BASIC and self._endpoint.password:
+                    self._endpoint.password = ""
```

Comment:
I agree with your point. But reason to remove password from endpoint is from security perspective. We don't want to keep password in memory once User is authenticated. User can make subsequent calls using Token set in cookies.